File: backend/TrainingInstitute.Api/Models/Student.cs
namespace TrainingInstitute.Api.Models;

// learner ki profile - email login ke liye User table me hi rakha hai
public class Student
{
    public int StudentId { get; set; }

    // User.UserId se jura hua - ek user ka ek hi student profile (unique)
    public int UserId { get; set; }

    public User? User { get; set; }

    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string? Phone { get; set; }

    public string? City { get; set; }

    public DateTime? DateOfBirth { get; set; }

    public string? AddressLine1 { get; set; }

    public string? AddressLine2 { get; set; }

    public string? GuardianName { get; set; }

    public string? EmergencyPhone { get; set; }

    public string?CollegeName { get; set; }

    public DateTime CreatedAt { get; set; }

    public int? CreatedBy { get; set; }
    public int? PassoutYear{get;set;}

    public DateTime? UpdatedAt { get; set; }

    public int? UpdatedBy { get; set; }

    // is student ne kaun kaun se course liye hai
    public ICollection<CourseEnrollment> Enrollments { get; set; } = new List<CourseEnrollment>();
}

File: frontend/src/types/student.ts
/*
 * Copyright (c) 2026 Anshul Negi
 * GitHub: https://github.com/NegiCoder
 * Unauthorized copying, modification, or distribution of this file
 * without explicit permission is prohibited.
 */

export type CreateStudentProfileRequest = {
  firstName: string
  lastName: string
  phone?: string | null
  city?: string | null
  dateOfBirth?: string | null
  addressLine1?: string | null
  addressLine2?: string | null
  guardianName?: string | null
  emergencyPhone?: string | null
  collegeName?: string | null
  passoutYear?: number | null
}

export type UpdateStudentProfileRequest = CreateStudentProfileRequest

export type StudentProfileResponse = {
  studentId: number
  userId: number
  fullName: string
  email: string
  firstName: string
  lastName: string
  phone?: string | null
  city?: string | null
  dateOfBirth?: string | null
  addressLine1?: string | null
  addressLine2?: string | null
  guardianName?: string | null
  emergencyPhone?: string | null
  collegeName?: string | null
  passoutYear?: number | null
  createdAt: string
  updatedAt?: string | null
}

export type StudentSearchRequest = {
  searchTerm?: string
  city?: string
  collegeName?: string
  passoutYear?: number
  pageNumber: number
  pageSize: number
}
